engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/AbstractTableLocationProvider.java

```diff
@@ -214,6 +245,31 @@ private void verifyPartitionKeys(@NotNull TableLocationKey locationKey) {
         }
     }
 
+    // region Mark / Sweep support
+    protected final void markLocationKeys() {
```

Comment:
It isn't  yet clear to me why we can't do this at our own (higher) level.

Reply:
The problem was the visitation.  The locations need to be visited by handleTableLocationKey,  which is final in AbstractTableLocationProvider.   I don't want to make it not final and override because then we'd have to copy-pasta the implementation. I could add a do-nothing protected method `onLocationAdded()` with some javadoc explaining how it is meant to be used.